### ISSUE TITLE
DockerExtension should allow string-based dependsOn

### ISSUE THREAD

**Issue by mnazbro** [palantir/gradle-docker#40]
I was quite confused to have to use real task objects rather than strings in the dependsOn field in the docker block. We should be able to use string-based input too just like a normal gradle task.

**Comment by uschi2000** [comment]
Do you have a minimal example that clarifies the inconsistency?

**Comment by mnazbro** [comment]
```
task bar() {
}
task foo() {
    dependsOn bar
}
```

and

```
task foo() {
    dependsOn 'bar'
}
task bar() {
}
```

vs

```
task foo() {
}
docker {
    name 'hub.docker.com/username/my-app:version'
    dependsOn foo
}
```

and

```
task foo() {
}
docker {
    name 'hub.docker.com/username/my-app:version'
    dependsOn 'foo'
}
```

**Comment by mnazbro** [comment]
It's important to note that the string version allows you to have not defined the task yet (and use a path to a task in a different project)

**Comment by uschi2000** [comment]
Hmm, it's odd that the docker plugin configures dependencies via the
DockerExtension. A more natural implementation would use task dependencies
instead.

**Comment by mnazbro** [comment]
The DockerExtension also does something extra though. It copies the output of each of those dependsOn tasks into the docker build folder.

**Comment by uschi2000** [comment]
Yep, I know. But really the task should do that.

**Comment by rmelick** [comment]
I agree.  Why is the docker task created by this plugin different from a normal gradle task, which would allow me to use the normal dependsOn?

I did figure out a workaround to allow referencing tasks by string, but it seems like there should be a better way to fix this.  You can use the tasks.getByPath method (see https://docs.gradle.org/current/userguide/more_about_tasks.html)

`dependsOn tasks.getByPath('bar')`

Any guidance on where the docker task is defined in the code?  I'm happy to send a pull request (although this would be the first gradle plugin I've worked on).

**Comment by markelliot** [comment]
We need/want to actually capture the task outputs when using `dependsOn` in the configuration. You can always grab the task and encode a dependency using something like `tasks.docker` to get a handle on it and add your dependencies.
